app/scripts/migrations/054.js

```diff
@@ -0,0 +1,71 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Remove tokens that don't have an address due to
+ * lack of previous addToken validation.  Also removes
+ * an unwanted, undefined image property
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+
+    const { PreferencesController } = versionedData.data;
+
+    if (PreferencesController.tokens) {
+      PreferencesController.tokens = PreferencesController.tokens.filter(
+        ({ address }) => address,
+      );
+    }
+
+    if (PreferencesController.hiddenTokens) {
+      PreferencesController.hiddenTokens = PreferencesController.hiddenTokens.filter(
+        Boolean,
+      );
+    }
+
+    if (PreferencesController.accountHiddenTokens) {
+      Object.keys(PreferencesController.accountHiddenTokens).forEach(
+        (address) => {
+          const networks = Object.keys(
```

Comment:
upon deeper dive, it might be slightly more correct to name this `chains` instead of `networks` because the keys are things like `0x1` versus `mainnet`. This is *purely* for the readability of the migration.